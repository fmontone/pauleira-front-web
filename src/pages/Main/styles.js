import styled from 'styled-components';

import PageWrapper from '~/components/PageWrapper';
import ContainerCustom from '~/components/ContainerCustom';

export const Wrapper = styled(PageWrapper)`
  border: 10px solid lime;
  /* background: linear-gradient(
    0deg,
    rgba(10, 0, 18, 1) 0%,
    rgba(68, 0, 127, 1) 100%
  ); */
`;

export const Container = styled(ContainerCustom)``;
