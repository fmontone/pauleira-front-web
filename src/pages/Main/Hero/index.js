import React, { useState, useEffect, useContext } from 'react';

import history from '~/services/history';
import { ActivePageContext } from '~/contexts/ActivePageContext';

import HeroIllustration from '~/assets/PauleiraHeroIsometricIllustration.png';
import { Container, Title, HeroImg, CallToAction } from './styles';

export default function Hero() {
  const courseList = [
    'Construção',
    'Regulagem',
    'Limpeza',
    'Customização',
    'Eletrônica',
  ];
  const [courseIndex, setCourseIndex] = useState(0);
  const { setActivePage } = useContext(ActivePageContext);

  useEffect(() => {
    const interval = setInterval(() => {
      setCourseIndex(
        courseIndex === courseList.length - 1 ? 0 : courseIndex + 1
      );
    }, 1600);

    return () => clearInterval(interval);
  }, [courseIndex, courseList]);

  function handleCallToActionClick() {
    setActivePage('cursos');
    history.push('/cursos');
  }

  return (
    <Container>
      <Title>
        Aprenda
        <span className="title__highlight">{courseList[courseIndex]}</span>
        de guitarras e contrabaixos
      </Title>

      <HeroImg
        src={HeroIllustration}
        alt="Pauleira Guitars Ilustração Isométrica da Luthier Paula Bifulvo ensinando técnicas de luthieria para um aluno"
      />

      <CallToAction onClick={handleCallToActionClick} width="stretch">
        Conheça Nossos Cursos
      </CallToAction>
    </Container>
  );
}
