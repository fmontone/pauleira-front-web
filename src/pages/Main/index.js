import React from 'react';

import NavBar from '~/components/NavBar';
import { Container } from './styles';

export default function Main() {
  return (
    <Container data-testid="pages-main">
      <NavBar />
    </Container>
  );
}
