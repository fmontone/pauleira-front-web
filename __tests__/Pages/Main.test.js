import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Main from '~/pages/Main';

describe('Pages/Main', () => {
  it('Should Render Main Page Without Crash', () => {
    const { findByTestId } = render(
      <BrowserRouter>
        <Main />
      </BrowserRouter>
    );

    expect(findByTestId('pages-main')).toMatchSnapshot();
  });
});
