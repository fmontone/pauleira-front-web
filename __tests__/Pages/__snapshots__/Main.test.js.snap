// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Main Should Render Main Page Without Crash 1`] = `Promise {}`;
